hazelcast/src/main/java/com/hazelcast/instance/DefaultNodeContext.java

```diff
@@ -41,7 +48,77 @@ public NodeExtension createNodeExtension(Node node) {
 
     @Override
     public AddressPicker createAddressPicker(Node node) {
-        return new DefaultAddressPicker(node.getConfig(), node.getProperties(), node.getLogger(AddressPicker.class));
+        Config config = node.getConfig();
+        AddressLocatorConfig addressLocatorConfig = config.getNetworkConfig().getAddressLocatorConfig();
+
+        if (!addressLocatorConfig.isEnabled()) {
+            return new DefaultAddressPicker(config, node.getProperties(), node.getLogger(AddressPicker.class));
+        }
+
+        AddressLocator implementation = addressLocatorConfig.getImplementation();
+        if (implementation != null) {
+            return new DelegatingAddressPicker(implementation, config.getNetworkConfig());
+        }
+        ClassLoader classLoader = config.getClassLoader();
+        String classname = addressLocatorConfig.getClassname();
+        Class<?> clazz = loadAddressLocatorClass(classLoader, classname);
+
+        Constructor<?> constructor = findAddressLocatorConstructor(clazz);
+        Properties properties = addressLocatorConfig.getProperties();
+        AddressLocator addressLocator = newAddressLocatorInstance(constructor, properties);
+        return new DelegatingAddressPicker(addressLocator, config.getNetworkConfig());
+
+    }
+
+    private AddressLocator newAddressLocatorInstance(Constructor<?> constructor, Properties properties) {
+        Class<?>[] parameterTypes = constructor.getParameterTypes();
+        Class<?> clazz = constructor.getDeclaringClass();
+        String classname = clazz.getName();
+        try {
+            if (parameterTypes.length == 0) {
+                //we have only the no-arg constructor -> we have to fail-fast when some properties were configured
+                if (properties != null && !properties.isEmpty()) {
+                    throw new IllegalStateException("Cannot find a matching constructor for AddressLocator.  "
+                            + "The address locator has properties configured, but the class " + "'" + classname
+                            + "' does not have a public constructor accepting properties.");
+                }
+
+                return (AddressLocator) constructor.newInstance();
+            } else {
+                if (properties == null) {
+                    properties = new Properties();
+                }
+                return (AddressLocator) constructor.newInstance(properties);
+            }
+        } catch (InstantiationException e) {
+            throw new IllegalStateException("Cannot create a new instance of AddressLocator '" + clazz + "'", e);
+        } catch (IllegalAccessException e) {
+            throw new IllegalStateException("Cannot create a new instance of AddressLocator '" + clazz + "'", e);
+        } catch (InvocationTargetException e) {
+            throw new IllegalStateException("Cannot create a new instance of AddressLocator '" + clazz + "'", e);
+        }
```

Comment:
duplicate code because of Java 6 support. nothing to do for the moment.

```
The following example, which is valid in Java SE 7 and later, eliminates the duplicated code:

catch (IOException|SQLException ex) {
    logger.log(ex);
    throw ex;
}
```
http://docs.oracle.com/javase/7/docs/technotes/guides/language/catch-multiple.html